api/src/controllers/contact.js

```diff
@@ -0,0 +1,52 @@
+const auth = require('../auth');
+const { Contact, Qualifier } = require('@medic/cht-datasource');
+const ctx = require('../services/data-context');
+const serverUtils = require('../server-utils');
+
+const getContact = ({ with_lineage }) => ctx.bind(with_lineage === 'true' ? Contact.v1.getWithLineage : Contact.v1.get);
+const getContactIds = () => ctx.bind(Contact.v1.getIdsPage);
+
+const checkUserPermissions = async (req) => {
+  const userCtx = await auth.getUserCtx(req);
+  if (!auth.isOnlineOnly(userCtx) || !auth.hasAllPermissions(userCtx, 'can_view_contacts')) {
+    return Promise.reject({ code: 403, message: 'Insufficient privileges' });
+  }
+};
+
+module.exports = {
+  v1: {
+    get: serverUtils.doOrError(async (req, res) => {
+      await checkUserPermissions(req);
+      const { uuid } = req.params;
+      const contact = await getContact(req.query)(Qualifier.byUuid(uuid));
+
+      if (!contact) {
+        return serverUtils.error({ status: 404, message: 'Contact not found' }, req, res);
+      }
+
+      return res.json(contact);
+    }),
+    getIds: serverUtils.doOrError(async (req, res) => {
+      await checkUserPermissions(req);
+
+      if (!req.query.freetext && !req.query.type) {
+        return serverUtils.error({ status: 400, message: 'Either query param freetext or type is required' }, req, res);
+      }
+      const qualifier = {};
+
+      if (req.query.freetext) {
+        Object.assign(qualifier, Qualifier.byFreetext(req.query.freetext));
+      }
+
+      if (req.query.type) {
+        Object.assign(qualifier, Qualifier.byContactType(req.query.type));
+      }
+
+      const limit = req.query.limit ? Number(req.query.limit) : req.query.limit;
```

Comment:
The reason why `req.query.limit` is being passed when the conditional is falsy is that in the `cht-datasource` there is already a validation for this and also a default value. This is to ensure that the validation does not happen twice and also the default value.
[Reference](https://github.com/medic/cht-core/pull/9311#discussion_r1717256585).

Reply:
Then why not have cht-datasource also do the Number conversion then? Why have this validation here? Is limit ever expected to _not_ be a number?

Reply:
> Is limit ever expected to not be a number?

Yeah, in cases like the one above, where it is passed as a query param in REST API, it is expected to be a stringified number. However, `cht-datasource` can also be used in non-REST API codes where end-users will have to pass in a number to `cht-datasource` because PouchDB expects the `limit` value to be a number. I think that's a reasonable approach to make the `limit` variable an explicit Number type, as that would align with the expected input for the PouchDB Adapter. This would provide better type safety and clarity in the code. The validation being present in `cht-datasource` still makes sense, and whether to apply the same validation elsewhere should be at the discretion of the end-user, based on their specific use case.

Reply:
So even if it's a stringified number or a number, we still only ever evaluate it as a number. so it makes sense to only have validation in one spot, right?

Reply:
Ok, I'll use a different word than validation: processing. 
It's possible to only do processing of the `limit` parameter in a single place: this includes validation, conversion and all the other things. We should do processing in a single place.

Reply:
In most cases, I agree but I'm not sure why the conversion of `limit` from `string` to `number` should be designated to `cht-datasource`, it's not its concern.

Reply:
I'm suggesting that conversion and validation can both be done in a single place: cht-datasource.

Reply:
@jkuester thoughts on this?

Reply:
Looking at this again, I cannot see any reason why it would be a problem to just directly pass `req.query.limit` to cht-datasource.  We already thoroughly validate the `limit` argument in the cht-datasource logic and JS is not going to have any issue auto-boxing any valid number (without needing to be wrapped in `Number()`)....

Reply:
:sweat_smile: :sweat_smile: :sweat_smile: I _knew_ we had discussed this before.  I was not able to find that thread you linked to and so I proceeded to repeat the exact same line of reasoning that lead me to start that thread in the first place... :facepalm: 

_sigh_

Now I think I am 100% following what you are saying above and agree with this statement:

> I'm not sure why the conversion of limit from string to number should be designated to cht-datasource, it's not its concern.

The cht-datasource apis are responsible for _sanitizing_ the input to ensure it confirms to the specified expectations.  Generally speaking, I do not think cht-datasource should need to include extra logic to "support" different ways that consumers decide to provide data (accidentally or intentionally).  The type `number | string` does not precisely communicate the valid range of values for `limit` (which should only ever be numeric).  

That being said, the cht-datasource interfaces should be designed to be convenient to consume as long as it does not compromise the clarity of the API. It turns out that TypeScript has a type that represents "a string containing a number value": `\`${number}\``! This means that we could specify the type of `limit` to be `number | \`${number}\`` and then handle converting any string to an actual number just inside the edge of the cht-datasource code flow.  To me, that seems like the best of both worlds where we are not always fighting the weird non-auto-boxing behavior when calling cht-datasource from JS code, but at the same time, our TS APIs do not have to be unnecessarily promiscuous.  Does that make sense?

Reply:
>  This means that we could specify the type of limit to be number | `${number}`

Sounds good. As long as we don't have logic around limit in multiple places. 
But, philosophically, I don't think a programming languages (typescript in this case) should dictate how software is written, and even if there wasn't a `type`, then we should still have the option to design software however we want. 

Please make the change.